prime-dotnet-webapi/Services/EnrolleeAgreementService.cs

```diff
@@ -124,6 +130,16 @@ public async Task<Agreement> GetCurrentAgreementAsync(int enrolleeId)
                 .FirstAsync(at => at.EnrolleeId == enrolleeId);
         }
 
+        public async Task<AgreementType> GetCurrentAgreementTypeAsync(int enrolleeId)
+        {
+            return await _context.Agreements
+                .Include(a => a.AgreementVersion)
+                .OrderByDescending(a => a.CreatedDate)
+                .Where(a => a.EnrolleeId == enrolleeId)
+                .Select(a => a.AgreementVersion.AgreementType)
+                .FirstAsync();
```

Comment:
This will throw if the given user has no `Agreements`, i.e. every user their first time through the app. This should be `.FirstOrDefault`